src/storage/src/hummock/sstable/builder.rs

```diff
@@ -340,6 +355,13 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
         }
 
         let table_id = full_key.user_key.table_id.table_id();
+        let vnode_id = full_key.user_key.get_vnode_id();
+
+        if let Some(vnode_bitmap_builder) = self.vnode_bitmap_builder.as_mut() && vnode_id != self.last_vnode {
+            vnode_bitmap_builder.set(vnode_id, true);
+            self.last_vnode = vnode_id;
+        }
```

Comment:
`last_vnode` is initialized with `0`, what if the incoming `vnode_id` is `0`, too? IIRC, RW's vnode id starts at 0.

Reply:
IMO, just setting the bitmap for each iter is enough. It doesn't cost a lot. (Maybe even lighter than conditioning.